waimak-core/src/main/scala/com/coxautodata/waimak/dataflow/spark/SparkInterceptors.scala

```diff
@@ -30,7 +30,7 @@ object SparkInterceptors extends Logging {
   def addPostCacheAsParquet(sparkFlow: SparkDataFlow, outputLabel: String)
                            (dfFunc: Dataset[_] => Dataset[_])
                            (dfwFunc: DataFrameWriter[_] => DataFrameWriter[_]): SparkDataFlow = {
-    def post(data: Option[Any], sfc: SparkFlowContext): Option[Dataset[_]] = {
+    def post(data: Option[Any], sfc: FlowContext): Option[Dataset[_]] = {
```

Comment:
You don't need sfc here?

You can use the flow context from the sparkflow object above then you don't need the cast below:

```scala
sfc.asInstanceOf[SparkFlowContext].spark.read.parquet(path)
```

becomes 
```scala
sparkFlow.flowContext.spark.read.parquet(path)
```

Reply:
Then we could change the definition of:

```scala
sealed case class CachePostAction[T](run: (Option[T], FlowContext) => Option[T], override val labelToIntercept: String) extends PostAction[T](labelToIntercept)

sealed case class TransformPostAction[T](run: (Option[T], FlowContext) => Option[T], override val labelToIntercept: String) extends PostAction[T](labelToIntercept)
```

to
```scala
sealed case class CachePostAction[T](run: Option[T] => Option[T], override val labelToIntercept: String) extends PostAction[T](labelToIntercept)

sealed case class TransformPostAction[T](run: Option[T] => Option[T], override val labelToIntercept: String) extends PostAction[T](labelToIntercept)
```


This is a biggish change, but i don't see why it wouldn't work? Would tidy things up a little, however it wouldn't give users access to the spark context if they wanted to define custom interceptors anymore. What do you think?